include/astshim/functional.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * LSST Data Management System
+ * Copyright 2017 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef ASTSHIM_FUNCTIONAL_H
+#define ASTSHIM_FUNCTIONAL_H
+
+#include "astshim/FrameSet.h"
+
+/*
+ * This header declares operations that treat ast objects (particularly Mapping
+ * and its subclasses) as functions to be manipulated.
+ */
+namespace ast {
+
+/**
+ * Construct a FrameSet that performs two transformations in series.
+ *
+ * When used as a Mapping, the FrameSet shall apply `first`, followed by
+ * `second`. Its inverse shall apply the inverse of `second`, followed by the
+ * inverse of `first`. The concatenation is only valid if `first.getCurrent()`
+ * and `second.getBase()` are equivalent.
+ *
```

Comment:
Let's leave the code as is, so that we have the full flexibility of AST.

I think the comment needs to be updated. Do the frames really have to be "equivalent" or can they just be "compatible", and what does that mean in either case?